src/typedefs.rs

```diff
@@ -579,6 +630,34 @@ macro_rules! construct_hmac_key {
         /// # Security:
         /// - __**Avoid using**__ `unprotected_as_bytes()` whenever possible, as it breaks all protections
         /// that the type implements.
+        ///
+        /// - The trait `PartialEq<&'_ [u8]>` is implemented for this type so that users are not tempted
+        /// to call `unprotected_as_bytes` to compare this sensitive value to a byte string. The trait
+        /// is implemented in such a way that the comparison happens in constant time. Thus, users should
+        /// prefer `SecretType == &[u8]` over `SecretType.unprotected_as_bytes() == &[u8]`.
+        /// Examples are shown below.
+        /// ```rust
+        /// use orion::hazardous::mac::hmac::SecretKey;
+        ///
+        /// // Initialize a secret key with 32 random bytes.
+        /// let key_bytes: [u8; 32] =
```

Comment:
Please see above comments about randomly generating keys. Again, here you can use `&[0u8; 128]` with a negated `assert` for comparison. This will also remove the long comment and use of `unprotected_as_bytes()`.

Please also see above comments about variable-time example and capitalisation.